src/mono/mono/mini/interp/mintops.def

```diff
@@ -6,19 +6,39 @@
  *
  * OPDEF (opsymbol, opstring, oplength (in uint16s), num_dregs (0 or 1), num_sregs, optype)
  * optype describes the contents of the instruction, following the dreg/sreg offsets.
+ * IROPDEF is like OPDEF but for instructions that are never executed by the interpreter,
+ *  meant for use by transform.c and other internals.
+ * WASMOPDEF are special opcodes that only run in WASM builds and can be ignored on other targets
  */
 
 /*
  * This file is parsed by genmintops.py to generate typescript during the wasm build process,
  *  so if you make any changes to its syntax you will need to update that script.
+ * Any preprocessing directives are removed without being parsed!
  */
 
-OPDEF(MINT_NOP, "nop", 1, 0, 0, MintOpNoArgs)
-OPDEF(MINT_NIY, "niy", 1, 0, 0, MintOpNoArgs)
-OPDEF(MINT_DEF, "def", 2, 1, 0, MintOpNoArgs)
-OPDEF(MINT_IL_SEQ_POINT, "il_seq_point", 1, 0, 0, MintOpNoArgs)
-OPDEF(MINT_DUMMY_USE, "dummy_use", 2, 0, 1, MintOpNoArgs)
-OPDEF(MINT_TIER_PATCHPOINT_DATA, "tier_patchpoint_data", 2, 0, 0, MintOpShortInt)
+// Compatibility shims for old code
+
+#ifndef IROPDEF
+#define __DEFINED_IROPDEF__
+#define IROPDEF(opsymbol, opstring, oplength, num_dregs, num_sregs, optype) OPDEF(opsymbol, opstring, oplength, num_dregs, num_sregs, optype)
+#endif // IROPDEF
+
+#ifndef WASMOPDEF
```

Comment:
I'm not sure I understand the utility of this def. Couldn't we just guard opcodes with normal platform ifdefs both in `mintops.def` and in `interp_exec_method` ? Seems more straightforward and maybe at some point we will have platform specific opcodes for other arches also.

Reply:
We can do that, yeah. It requires genmintops to understand ifdef though potentially